projects/igniteui-angular/src/lib/core/utils.ts

```diff
@@ -579,14 +580,113 @@ export const isConstructor = (ref: any) => typeof ref === 'function' && Boolean(
  * Similar to Angular's formatDate. However it will not throw on `undefined | null | ''` instead
  * coalescing to an empty string.
  */
-export const formatDate = (value: string | number | Date, format: string, locale: string, timezone?: string): string => {
+export function formatDate(value: Date | string | number | null | undefined, format: string, locale: string, timezone?: string): string {
     if (value === null || value === undefined || value === '') {
         return '';
     }
-    return _formatDate(value, format, locale, timezone);
-};
+    if (typeof value === "string" || typeof value === "number") {
+        value = new Date(value);
```

Comment:
Using `new Date(string)` is most probably wrong for ISO dates.
See issue #15181 for details.

Reply:
With "regression" I mean this:
<img width="1215" height="702" alt="image" src="https://github.com/user-attachments/assets/98ac7a0b-ea95-4356-8286-8cf0a64678a2" />
both from the same data: '2025-08-01'

Reply:
Yes, that is because Angular handles the date formatting themselves and the Date object on its own is not enough. That is the whole point of the Intl we are using now.
I personally am ok with this breaking change, because I do not think previously the default behavior of Angular is the better one. There are a lot of discussion what is the "correct" behavior, but to me the date displayed should by default be in local timezone first and if the dev wants to specify target timezone should set it anyway.

Reply:
> There are a lot of discussion what is the "correct" behavior.

I don't think there's anything to discuss 2025-01-01 should NEVER be formatted as 31 Dec 2024.

Reply:
Like I said the Angular pipes and method are cosmetic and purely frontend oriented while the grid data object provided is supposed to be a backend one. These are quite different apples and oranges and we cannot apply the same logic.

Reply:
If we are to support the whole range of ISO standard though with the newest things, that will require quite a lot of other things though.